pathfinding/core/grid.py

```diff
@@ -95,7 +101,7 @@ def neighbors(self, node, diagonal_movement=DiagonalMovement.never):
                 neighbors.append(self.nodes[y - 1][x])
                 s0 = True
         # →
-        if x == self.width and self.passable_left_right_border:
+        if x == self.width - 1 and self.passable_left_right_border:
```

Comment:
Nice catch with the -1 (self.width is the length and we are accessing array indices, so this might raise an IndexError)! I think this bug was not seen before? Good bugfix.

Reply:
Think I made the bug myself and then fixed it :D Thanks anyway!

Reply:
Oh oups... Anyway, thanks for the pull request!